packages/eslint-plugin/configs/recommended.js

```diff
@@ -1,7 +1,27 @@
-module.exports = {
+/**
+ * WordPress dependencies
+ */
+const defaultPrettierConfig = require( '@wordpress/prettier-config' );
+const { hasPrettierConfig } = require( '@wordpress/scripts-utils' );
+
+const config = {
 	extends: [
 		require.resolve( './recommended-with-formatting.js' ),
 		'plugin:prettier/recommended',
 		'prettier/react',
 	],
 };
+
+if ( ! hasPrettierConfig() ) {
+	config.rules = {
+		'prettier/prettier': [
+			'error',
+			defaultPrettierConfig,
+			{
+				usePrettierrc: false,
+			},
+		],
+	};
+}
```

Comment:
I see what it is that we're trying to address, but something doesn't sit very well with me that the configuration is generated dynamically based on factors from the environment (directory) from which the linting is run. Is there any prior art to this?

Reply:
How about we simply make this ESLint rule always present with this custom WordPress Prettier config set? If someone doesn’t like it, they can use a different ruleset offered or simply override it.

In fact, I could do the opposite in the `wp-scripts lint-js` to what this PR tries and override the rule only when the Prettier config is detected in the project.

Reply:
> How about we simply make this ESLint rule always present with this custom WordPress Prettier config set? If someone doesn’t like it, they can use a different ruleset offered or simply override it.

It seems reasonable to me. As I consider it more, I think it's pretty sensible that these rules assume a default that the Prettier rules be configured to the WordPress standard. I'm pretty okay if we keep consideration of an override configuration from the local directory, if it helps avoid a breaking change.

Reply:
>How about we simply make this ESLint rule always present with this custom WordPress Prettier config set?

Would this mean we wouldn't need `hasPrettierConfig` here anymore? Which I think is largely the primary purpose of the proposed new package, in making this utility available? Per https://github.com/WordPress/gutenberg/pull/21503#issuecomment-612893121, I think I wouldn't really mind some of these utilities still. I'd be happier if we didn't have `hasBabelConfig`, `hasJestConfig`, `hasPrettierConfig`, and `camelCaseDash` on the public API though.

Reply:
> I think it's pretty sensible that these rules assume a default that the Prettier rules be configured to the WordPress standard.

> Per [#21503 (comment)](https://github.com/WordPress/gutenberg/pull/21503#issuecomment-612893121), I think I wouldn't really mind some of these utilities still. I'd be happier if we didn't have `hasBabelConfig`, `hasJestConfig`, `hasPrettierConfig`, and `camelCaseDash` on the public API though.

It all makes sense. Let's do it and open another PR that addresses the related issue with the default coding styles set for Prettier in ESLint.

Reply:
Prettier + ESLint integration was tackled in https://github.com/WordPress/gutenberg/pull/21602.

I updated this PR according to the feedback received. It's ready for another check.